templates/_components/image.twig

```diff
@@ -0,0 +1,96 @@
+{# @var \craft\elements\Asset|null image #}
+{% set image = image ?? null %}
+
+{# @var string|null class #}
+{% set class = class ?? null %}
+
+{#
+  @var string|null loading
+  Can be 'lazy' or 'eager'
+#}
+{% set loading = loading ?? 'lazy' %}
+
+{#
+  @var array sizes
+
+  The 'sizes' prop is used to specify the size of the image at different viewport widths.
+  It should be an associative array where:
+  - Keys are either Tailwind breakpoint names (sm, md, lg, xl, 2xl) or custom media queries.
+  - Values are CSS length values (e.g., px, em, rem, vw) representing the image width at that breakpoint.
+
+  Example usage:
+  sizes:  {
+    default: '100vw',
+    sm: '50vw',
+    md: '33vw',
+    '(min-width: 1200px)': '400px'
+  }
+  If not provided, it defaults to full viewport width at all sizes.
+
+#}
+{% set sizes = sizes ?? {
+  default: '100vw',
+} %}
+
+{#
+  Calculated by dividing width by height.
+  @var float|null ratio
+#}
+{% set ratio = ratio ?? null %}
+
+{% set defaultSize = sizes.default ?? '100vw' %}
+
+{#
+  Use Tailwind screen config in our sizes prop.
+  Keep this in sync with your TW's screens config.
+  https://tailwindcss.com/docs/screens
+#}
+{% set screens = {
+  'sm': '(min-width: 640px)',
+  'md': '(min-width: 768px)',
+  'lg': '(min-width: 1024px)',
+  'xl': '(min-width: 1280px)',
+  '2xl': '(min-width: 1536px)',
+} %}
```

Comment:
On some sites (like MW), we actually convert our TW config to JSON and then load it via PHP so we can get at these values. 

But.... I'm of the impression lately that it a ton of complexity to save us 5-ish lines of code. 

Do screen sizes really change all that much after a project has been started? This might be an acceptable `DUPE`